src/providers/BasicPoolsProvider.tsx

```diff
@@ -111,6 +153,158 @@ export default function BasicPoolsProvider({
   )
 }
 
+/**
+ * WILL NOT BE USED UNTIL POOLS_MAP and TOKENS_MAP have been fully deprecated in other pages
+ * Retrieve PoolData from Registry
+ * excludes outdated Pools
+ */
+/**
+ * Get all info about pools that can be read on chain.
+ * Excludes price data and user data.
+ * Will be replaced with registry.
+ */
+export async function getPoolsDataFromRegistry(
+  chainId: ChainId,
+  poolRegistry: PoolRegistry,
+  poolRegistryMultiCall: MulticallContract<PoolRegistry>,
+  ethCallProvider: MulticallProvider,
+): Promise<DeepNullable<SwapInfo>[]> {
+  const poolCount = (await poolRegistry.getPoolsLength()).toNumber()
+  const registryPoolDataMultiCalls = enumerate(poolCount).map((index) =>
+    poolRegistryMultiCall.getPoolDataAtIndex(index),
+  )
+
+  const registryPoolData = await ethCallProvider.tryAll(
+    registryPoolDataMultiCalls,
+  )
+
+  const arePoolsPausedMulticalls: MulticallCall<unknown, boolean>[] = []
+  const virtualPricesMulticalls: MulticallCall<unknown, BigNumber>[] = []
+  const swapStoragesMulticalls: MulticallCall<unknown, SwapStorage>[] = []
+  const aParametersMultiCalls: MulticallCall<unknown, BigNumber>[] = []
+  const tokenBalancesMultiCalls: MulticallCall<unknown, BigNumber[]>[] = []
+  const underlyingTokenBalancesMultiCalls: MulticallCall<
+    unknown,
+    BigNumber[]
+  >[] = []
+  const lpTokenTotalSuppliesMultiCalls: MulticallCall<unknown, BigNumber>[] = []
+
+  registryPoolData.forEach((poolData) => {
+    if (poolData != null) {
+      const { lpToken, poolAddress } = poolData
+      const lpTokenContract = createMultiCallContract<Erc20>(
+        lpToken.toLowerCase(),
+        ERC20_ABI,
+      )
+
+      lpTokenTotalSuppliesMultiCalls.push(lpTokenContract.totalSupply())
+      arePoolsPausedMulticalls.push(
+        poolRegistryMultiCall.getPaused(poolAddress),
+      )
+      virtualPricesMulticalls.push(
+        poolRegistryMultiCall.getVirtualPrice(poolAddress),
+      )
+      swapStoragesMulticalls.push(
+        poolRegistryMultiCall.getSwapStorage(poolAddress),
+      )
+      aParametersMultiCalls.push(poolRegistryMultiCall.getA(poolAddress))
+      tokenBalancesMultiCalls.push(
+        poolRegistryMultiCall.getTokenBalances(poolAddress),
+      )
+      underlyingTokenBalancesMultiCalls.push(
+        poolRegistryMultiCall.getUnderlyingTokenBalances(poolAddress),
+      )
+    }
+  })
+
+  const [
+    arePoolsPaused,
+    virtualPrices,
+    swapStorages,
+    aParameters,
+    tokensBalances,
+    underlyingTokensBalances,
+    lpTokensSupplies,
+  ] = await Promise.all([
```

Comment:
Complete data about an arbitrary number of pools in 8-ish calls, amazing :heart_eyes: